### ISSUE TITLE
Rancher Dashboard Missing Secure-Boot and TPM for Harvester Provisioned Clusters

### ISSUE THREAD

**Issue by deiberts86** [rancher/dashboard#12920]
**Setup**
- Rancher version: 2.9.3
- Rancher UI Extensions: Virtualization Manager (Built-in 1.0.0)
- Browser type & version: Google Chrome version 109.0.5414.119 (Official Built) (arm64)

**Describe the bug**
Checkboxes to turn on secure boot and TPM capabilities are missing from Harvester Provisioned clusters within the Rancher MCM > Cluster Management > Create > Harvester > Infrastructure section.

**To Reproduce**
1. Install Rancher MCM
2. Add Harvester Cluster to Rancher MCM
3. Add Cloud Credentials for Harvester Cluster
4. Create Harvester Provisioned Cluster
5. For the Infrastructure, click on the advanced section to add your userData and the only checkbox is install qemu-agent. Should include TPM and secure-boot.

**Result**
SecureBoot and TPM checkboxes are missing for creating Harvester Cluster under the Advanced, UserData cloud-init box.  

**Expected Result**
SecureBoot and TPM checkboxes should be there if the underlying Harvester System supports it. 

**Screenshots**
![Screenshot 2024-12-19 at 3 46 19 PM](https://github.com/user-attachments/assets/b318ff4a-bdfd-48d0-926f-26f2d6f5debb)

<img width="1496" height="737" alt="Image" src="https://github.com/user-attachments/assets/a0e95acf-b144-4e2a-bbde-0f138460aae0" />

![Secure-boot-and-TPM_Rancher](https://github.com/user-attachments/assets/44d13162-ab11-4913-95ba-00998de8caf3)

**Additional context**
For DoD or other customers, UEFI secure-boot for VMs while using TPM should be something that users can turn on and off from within the Rancher UI while leveraging Harvester provisioned RKE2 or k3s clusters. Very similar to what vSphere provided clusters can do.

**Comment by richard-cox** [comment]
@a110605 what's the general process for harvester ui issues now, should it stay here or be transferred to a repo in the harvester org (like https://github.com/harvester/harvester-ui-extension)?

**Comment by CamrynCarter** [comment]
SURE-9578

**Comment by a110605** [comment]
> @a110605 what's the general process for harvester ui issues now, should it stay here or be transferred to a repo in the harvester org (like https://github.com/harvester/harvester-ui-extension)?

Please open the issue in https://github.com/harvester/harvester/issues. Harvester team manager Kiefer will triange and assign to related engineer.

**Comment by richard-cox** [comment]
Doesn't look like i can transfer this across orgs. @deiberts86 can you create this issue in https://github.com/harvester/harvester/issues, until then @a110605 i've assigned to you to own

**Comment by deiberts86** [comment]
Yes, I'll add an issue on the Harvester side of the house. Sorry, missed this reply. I'll make another reply with the issue number ref'ed here.

**Comment by deiberts86** [comment]
Issue on the Harvester side found here:
https://github.com/harvester/harvester/issues/7365

**Comment by a110605** [comment]
Reopen this issue as the requirement https://github.com/harvester/harvester/issues/7365 needs to be implemented in rancher/dashboard. We can plan it for v2.13.0. cc @nwmac , @richard-cox.

**Comment by izaac** [comment]
Manual validation required.

**Comment by a-arias** [comment]
TESTS
Rancher : v2.13-393c2a0e781a5cb24b8187eaf91bad51595b8b85-head
UI: master 9df831c

Fix validated ✅

https://github.com/user-attachments/assets/0ccee99f-d38c-4c74-a4a7-557b6d0757ed
